lib/BridgedRoom.js

```diff
@@ -0,0 +1,123 @@
+"use strict";
+
+function BridgedRoom(bridge, gitter, gitterRealtime, matrixRoomModel, gitterRoomModel) {
+  this._bridge = bridge;
+  this._gitter = gitter;
+  this._gitterRealtime = gitterRealtime;
+  this._matrixRoomModel = matrixRoomModel;
+  this._gitterRoomModel = gitterRoomModel;
+}
+
+BridgedRoom.prototype.matrixRoomId = function() {
+  return this._matrixRoomModel.getId();
+};
+
+BridgedRoom.prototype.gitterRoomName = function() {
+  return this._gitterRoomModel.getId();
+};
+
+BridgedRoom.prototype.joinAndStart = function() {
+  // We have to find out our own gitter user ID so we can ignore reflections of
+  // messages we sent
+  return this._bridge.getMyGitterUserId().then((gitterUserId) => {
+    this._gitterUserId = gitterUserId;
```

Comment:
Style: **Please** don't do this. I'm trying to form a mental model of all the variables in play in this class so I can see what is going on. It is extremely valuable to be able to just look at the constructor and see which variables are available. It is highly obfuscating to assign things to `this` *after* construction like this because:
 - A person writing a new method may miss this bit and then do the work again to extract the gitter user ID.
 - A person may assume that the gitter user ID is always set and not null guard.

I would strongly advocate being upfront in your constructor:
```javascript
function BridgedRoom(opts) {
  this._gitterUserId = null;
}
```

Bonus points if this is documented with what it will contain. This also has a nice debugging side-effect of assigning this field to `null`, rather than `undefined` (if it is not defined at all). This is nice because when you see JS exceptions they will say if "undefined is not a function" or "null is not a function". This allows you to quickly identify if say you typod `_giterUserId` (which will be `undefined`) or tried to access it before it was set (which will be `null`).